### PR TITLE
deluge:  1.1.11 -> 2.0.3

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Requires libtorrentrasterbar update, will link relevant PR's shortly.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
#64496 for dependency bump, cc'ing recent deluge PR's that we might want to coordinate /test together, or even just encourage seeing these all through! :)

#64112 .. i thought there was at least one other, teehee.

**Comment by dtzWill** [comment]
@GrahamcOfBorg build deluge
@GrahamcOfBorg test deluge

(I'm unsure if the test currently passes or is obviously going to fail due to changes across major version, but let's find out! ^_^)

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2464,7 +2464,8 @@ in
   ddrutility = callPackage ../tools/system/ddrutility { };
 
   deluge = callPackage ../applications/networking/p2p/deluge {
-    pythonPackages = python2Packages;
+    pythonPackages = python3Packages;
+    libtorrentRasterbar = libtorrentRasterbar.override { python = python3; };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: probably better to perform this override inside the deluge expression to ensure the same python is always used.

Comment (FRidh) [reply pkgs/top-level/all-packages.nix]: Actually, this package is already overridden in `python-packages.nix`!

**Comment by petabyteboy** [comment]
Thanks a lot! Quite a few of the trackers I'm using haven't whitelisted the new version yet and probably won't do so very soon because of the big changes it introduces. I would propose having deluge and deluge2 coexist for some time until it is accepted by most trackers.

**Comment by Infinisil** [comment]
Yeah agreed with @petabyteboy 

I also just heard that Deluge 2 apparently performs auto-migration of its state, so you can't easily downgrade again. So this should definitely be separate from v1 for a while

Ping @bricewge for maybe some help with getting the NixOS module to work

**Comment by peterhoeg** [comment]
> Quite a few of the trackers I'm using haven't whitelisted the new version yet

This might be off-topic, but do you know how that happens @petabyteboy? Is it similar to the user-agent in a browser or is there something else going on?

**Comment by peterhoeg** [comment]
@dtzWill, please note that the upgrade breaks the nixos module.

**Comment by petabyteboy** [comment]
> > Quite a few of the trackers I'm using haven't whitelisted the new version yet
> 
> This might be off-topic, but do you know how that happens @petabyteboy? Is it similar to the user-agent in a browser or is there something else going on?

Yes, the client sends a 20 byte peer id containing implementation and version information: http://www.bittorrent.org/beps/bep_0020.html

**Comment by mweinelt** [comment]
Hey, I just found it and needed to test it. So it works, but it is missing icons everywhere.

This is what happens at startup:
```
(deluge:22236): Gtk-WARNING **: 23:38:11.963: Could not find the icon 'list-remove-symbolic-ltr'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
	http://icon-theme.freedesktop.org/releases
```

**Comment by evanjs** [comment]
Icons seems to appear fine outside of a `nix-shell`, presumably because `hicolor-icons-theme` is in my `systemPackages`, but within a `nix-shell`, many of the icons are indeed missing.  I am not seeing any warning messages, though.

Additionally, this behavior does not seem to be uncommon among GTK apps, so I wonder if somebody with more experience in this realm has some thoughts on the matter.

**Comment by Soundtoxin** [comment]
Any news on this? I would love to be able to update soon!

**Comment by petabyteboy** [comment]
I would also like to update, but I would need to be able to run 1.x and 2.x in parallel because my trackers still haven't whitelisted it *sigh*.
I will prepare a PR to make that possible.

**Comment by Infinisil** [comment]
To be superseded by above PR

**Pull request closed without merging**